### DISCUSSION TITLE
Stepper

### DISCUSSION THREAD

**Discussion by annefranca** [Telefonica/mistica-design#1434] [Proposals]
### Goal

Discutir e avaliar a proposta de um novo componente Stepper ao sistema de design e determinar qual workflow esse componente deve seguir.

### Context

Precisamos de um componente novo de Stepper que seja consistente e reutilizável para que os times não apliquem soluções próprias em cada fluxo do App Vivo.

### Impact

Adicionar esse componente em uma das bibliotecas melhoraria a consistência na experiência da pessoa usuária.

### Discussion Questions

Esse componente ficaria em Mística ou em Extended?

### Do you have a visual proposal?

https://www.figma.com/file/4kwrFyZ5h9gi4vJjZiQbgR/Stepper?type=design&node-id=0%3A1&mode=design&t=jZc8b2XOmpExRBND-1

**Comment by yceballost** [comment]
it would enable the stepper component to have more use within other elements.

I see two ways:
1. a prop to hide the stepper icons 2.
2. Add another type of stepper for these cases

Some doubts that I have, what happens with the labels of the stepper on desktop? The current stepper show label in desktop views

**Reply by annefranca** [reply]
@yceballost não entendi os caminhos que você pontuou (1 e 2). Não temos ícones no componente Stepper.

Nesse caso, esse Stepper não teria rótulos no componente em si, mas poderia ter um detalhe sobre o mesmo na interface. Essa é a forma que estamos implementando as etapas detalhadas em no detalhe da página https://www.figma.com/file/4kwrFyZ5h9gi4vJjZiQbgR/Stepper?type=design&node-id=2103%3A14648&mode=design&t=8RUCdyGOGRgKlxZI-1

**Comment by annefranca** [comment]
@aweell podemos dar continuidade nessa discussion?

**Reply by aweell** [reply]
Hi @annefranca I have some doubts regarding this proposal

After reviewing the examples you're providing whats the difference between that stepper and A progress bar?

![Screenshot 2023-09-14 at 11 46 29](https://github.com/Telefonica/mistica-design/assets/44420072/ee74adf3-8549-4620-870d-5f0f5e575e8d)

Don't both components report the same: a change in progress?

From what I understand the difference here is that there are a series of important milestones that will be shown in the detail page. With the current progress bar is impossible to directly define milestones in the component, is a straight path from A to B.

There's other alternatives you have explored? I'm thinking for example:

[Example: Progress tracker using a progress bar and steps](https://mistica-web.vercel.app/playroom/#?code=N4Igxg9gJgpiBcJgApkEoAEBeAfB4AOgHYYYBuAhgE7kUA2ArjNhgOYwAuAyhxRzMgIhKjGEIA0GAEwBWNAG5ipSjQDOnAGr0mLdd179Bw7WJALiSjJCKqOGWzAAOLdNjyESpDAEsAZhmQRHQAeLGk5fEsvDCpOBioSD2joh0d4DABGcSjkjm8OOhh0oS4YKjJvAHOIDBg6AEuOKm9IDAp2IigKKAgJHOjYVTBmxzyIInh%2B5JKGCnJvVXy5jnqwIha59SoAQ4wiTZhWBk6KSQAGAE4AekvJGCIm5gyzgAtazKkXvs9ogF9FH4YX61OjqHz%2BQImDB4WQYABkcNoogwoQwMjOmCSyViHHiiSmpFS6Sk2UB0TyBSKGBKZQq1RBjWarTmYAoAFtvEQXr0QKTkl5BsNvKNvONJmTpiAAKJsjAAI1iZGYNRWaxaNQAYts5aKrC9DtRdhQauoGLVOmUYNV0gAlAB0GAAUhAAMc1ACyEHipwwAA4AOwXb78-45YF1MF%2BAJBZg4MLo%2BGImMosL%2BiJY6I4vGRCWE-hpDAAZj5-IwFMKxRApXKVRqdUZ6r2NToEFZdGDpcFIzGEwJXiEAHky6t1q0tTqamB9awvU2MC22xhHNQ5hQAFYMLo0TYQRhgUX7GgmmBmxxUCBywpsigdv4Av4gyMQ5NxjBpzFTLMJHOl%2Bz59IACwlvy5ZUkIADCLwUEq3jnlY4xgIwAC3PS3skXbCj24q-qQNKnuel4wNeGC%2BBA3gxDAqi7hUPTvJRvBshAqgYAABxgUALI4TFVG6S4rhgACOsx0EJZTsQAXzRq4cfu4z0GhpChoCvyWEpljWLYVjxLEDw8E4uj5ugAKWF%2BJCCICwQ2pRXE2N4SoADIUAAnl6HA4FMwQ8BQYAANb2MuYAwFgwCFlIvzuRKwQACJ8BQ4HUFAfagVgQgAIKqIstgAFdEPucxSABLxUKoQh9jAAAeTQUMFfakMEABCEDlXxUAcUQrAACoQI4wUZAAbOFtVeJ5vC%2Bf53lBcA-WDbmpYjd5fmqAFk0ATNOFzR1FUcFIOB6QWwBgNp9z6E4dqpMCsAYABwRXJtlU7UN-LBHdHCFjgB1Hbp%2BZ2qBvw3S9b2Pckz1bVI8EtlQwWQBDqg-VtpTWFuTlretpYfVQOknY4doYSK4wqbNv7-aDEWo8NVxeb5pNkyiAAK56sLEGX1dQS4M0zqi02UgUPMFMbAlc1PrTdlM%2BULc1XI15Xi38UyCx5ACSRB0JyzBLRNwW%2BjNfYNQwHAcOMCPjEjQ2qNedDtoTGDjPTlGqMFri4D%2BqMvuEGAAPz2JoJiQsiAC04SYOkehaKIvtMOYVu-ATv4y8ktsVF6zGpDrkt6wbRBGycVBOXHdX1enhswIj1BOab5uW%2BtNscw7mBOxmOHJsEmRnGcHtexwodMOHzAANSBxgwfe2HMaR%2Bt0d9nnXgAHJbX%2BTipwX%2BtFyXOdxzdSsq0QMBTyLo1izkN1WUt4yLA5zmudTke-Gg6AE7yIAcPqbKUQgADaIAaHZEAAPqzwA7iAAAuuIEA-9vBQCfqod%2BhY%2BpnCAb8IAA)


In this case, the milestones are directly expressed in a text above the current stage.


If you still believe your proposal makes more sense, I doubt the stepper is the component we need to modify. Instead I strongly bet for creating a new stepped variant of the progress bar so the UI proposed can be achieved.

![Screenshot 2023-09-14 at 13 45 02](https://github.com/Telefonica/mistica-design/assets/44420072/4238e374-9043-4976-b16d-b27bcf7060f2)

**Reply by annefranca** [reply]
@aweell a diferença são os passos importantes de cada etapa, nesse caso, o componente Progress Bar, acaba não transmitindo essa essência desse comportamento. Vale ressaltar que podemos ter mais de 4 etapas dentro de um processo. 

Importante: esse componente também pode ser aplicado em fluxos onde a pessoa usuária precisa passar por mais de uma etapa e queremos indicar onde a pessoa usuária se encontra e quantas etapas ainda faltam para finalizar um determinado processo, segue exemplo: https://www.figma.com/file/4kwrFyZ5h9gi4vJjZiQbgR/Stepper?type=design&node-id=2230%3A4709&mode=design&t=DPrx75O2V0vef7ll-1

**Comment by yceballost** [comment]
# Evaluation
There was indeed a lot of debate about the differences between stepper and progressBar for the Vivo case.

---

Various scenarios have been evaluated:

## 1. ProgressBar stepped as a variant
Adding a prop to manage steps in the current progress and changing the component's appearance to display X progress bars with their behavior.

## 2. Stepper variant without icons
The possibility of adding the option for the stepper to have no icons and achieve the same result was proposed. However, it is dismissed because the stepper is considered to provide additional information to the user about the next steps through labels, numbers, and icons of the completed steps. This is useful in certain product processes or screens, and we prefer to keep them as components of the same family but separate.

## 3. Merge ProgressBar & Stepper into a type of component "Progress" (ProgressBar, ProgressTracker)
The question of why there is a difference between the two components is raised. The idea is for the components to have an associated use and prevent designers from using the components incorrectly. However, in the case of stepper vs. progressBar, it can be understood that there are scenarios where you want to show progress without the need to emphasize it with a stepper.

To allow more freedom to UX in making these decisions, it is suggested that everything be a "Progress" component (with its variants) and eliminate the differentiation between stepper and progress.

---

# Solution
We will go with scenario #1. We will include a variant in the progressBar to cover the case of having a progressBar with steps. From scenario #3, we will consider the idea of grouping progressBar and stepper under the "Progress Indicators" category.

For the multi-step flow demonstrated by Vivo, the use of the progressBar stepped could also be valid.

**Reply by annefranca** [reply]
@yceballost consegue me mostrar uma proposta visual da sua solução? Dentro do que você está trazendo vamos ter possibilidade da aplicação do componente "small" para aplicar nos fluxos de etapas como demostrado no arquivo que compartilhamos com vocês?

**Reply by aweell** [reply]
The component should take the 100% of the width of the parent so, if "small" refers to decrease the width of the component it would be possible just decreasing the width of the parent wrapper. Otherwise making changes in the component height doesn't make any sense.

Each step will take the: `width` / `number of steps` - `the sum of space between steps`

**Reply by annefranca** [reply]
@aweell perfeito! Seria na largura mesmo. Fico no aguardo dessa proposta visual.

**Reply by aweell** [reply]
It will behave practically like this example:

https://tinyurl.com/27yp3u9f

**Reply by annefranca** [reply]
@aweell muito bom! Você tem uma previsão evolução do componente estar disponível? Pergunto para poder alinhar uma data com o time da Vivo.

**Reply by yceballost** [reply]
Tenemos release esta semana y tenemos bastante tareas antes del cierre del quarter pero la definición de diseño estará lista para la semana próxima (aunque intentaremos cerrarla antes). Lo dije por https://jira.tid.es/browse/OBVIVO-1998 igualmente
